### PR TITLE
Implementing train pattern

### PR DESCRIPTION
**Pull request by arhamahmed**
@irfansharif

### CODE REVIEW

File: lib/cerebrum/cerebrum.rb [hunk 1]
```diff
@@ -1,8 +1,7 @@
 require_relative "data_scrubber"
 require_relative "helper"
 require_relative "lin_alg"
-
-require "pp" #for slightly better logging [temporary]
+require "pp"
```
Comment (irfansharif) [inline lib/cerebrum/cerebrum.rb:4]: we should make sure to remove this later.

File: lib/cerebrum/cerebrum.rb [hunk 2]
```diff
@@ -138,9 +136,10 @@ def calculate_deltas(target)
 
   def run_input(input)
     @outputs[0] = input
+    output = []
```
Comment (irfansharif) [inline lib/cerebrum/cerebrum.rb]: look at the diff generated, this was not merged correctly. `output = []` was removed, this is no longer the return statement but `@outputs.last` is which was moved down to [here](https://github.com/irfansharif/cerebrum/pull/10/files#diff-fd735047d6442f7f9ab1c85395a69b3cR205) for you.

File: test/cerebrum_test.rb [hunk 3]
```diff
@@ -23,6 +23,7 @@ def test_train
     ]
 
     @network = Cerebrum.new
-    @network.train(scrubbed_dataset)
+    result = @network.train(scrubbed_dataset)
+    p "result: #{result}"
```
Comment (irfansharif) [inline test/cerebrum_test.rb:27]: you missed a debug statement ☝️

Comment (arhamahmed) [inline test/cerebrum_test.rb:27]: leaving that one there for now until I know what the actual result should be

File: lib/cerebrum/cerebrum.rb [hunk 4]
```diff
@@ -149,16 +148,64 @@ def run_input(input)
...
-          sum += @outputs[layer - 1][prev_node] * weights[prev_node] 
+          sum += @outputs[layer - 1][prev_node] * weights[prev_node]
         end
         @outputs[layer][node] = activation_function(sum)
       end
     end
+    output
+  end
+
+  def mean_squared_error(errors)
```
Comment (irfansharif) [inline lib/cerebrum/cerebrum.rb]: you actually have this function defined twice.

File: lib/cerebrum/cerebrum.rb [hunk 5]
```diff
@@ -149,16 +148,64 @@ def run_input(input)
...
+    sum = 0
+    errors.each do |error|
+      sum += error * error
+    end
+    sum / Float(errors.length)
+  end
+
+  def adjust_weights(rate)
```
Comment (irfansharif) [inline lib/cerebrum/cerebrum.rb:118]: ditto for this and `calculate deltas`. otherwise LGTM 👌

File: lib/cerebrum/cerebrum.rb (hunk 4 above)
Comment (arhamahmed) [inline lib/cerebrum/cerebrum.rb]: this happened after the first rebase, as well... strange

**Pull request merged**